backend/open_webui/functions.py

```
@@ -0,0 +1,316 @@
+import logging
+import sys
+import inspect
+import json
+
+from pydantic import BaseModel
+from typing import AsyncGenerator, Generator, Iterator
```

Ruff (0.8.2)

Import from `collections.abc` instead: `AsyncGenerator`, `Generator`, `Iterator`

Import from `collections.abc`

(UP035)